src/control/server/storage/bdev/backend.go

```diff
@@ -179,41 +179,39 @@ func (sb *spdkBackend) prepare(req storage.BdevPrepareRequest, userLookup userLo
 		return resp, nil
 	}
 
-	// If VMD has been explicitly enabled and there are VMD enabled NVMe devices on the host,
-	// attempt to prepare them first.
-	vmdReq, err := getVMDPrepReq(sb.log, &req, vmdDetect)
-	if err != nil {
-		return resp, errors.Wrapf(err, "create vmd request")
+	// Update request if VMD has been explicitly enabled and there are VMD endpoints configured.
+	if err := updatePrepareRequest(sb.log, &req, vmdDetect); err != nil {
+		return resp, errors.Wrapf(err, "update prepare request")
 	}
-	if vmdReq != nil {
-		if err := sb.script.Prepare(vmdReq); err != nil {
-			return resp, errors.Wrap(err, "re-binding vmd ssds to attach with spdk")
+	resp.VMDPrepared = req.EnableVMD
+
+	if !req.EnableVMD {
+		// Before preparing, reset bindings of NVMe devices. If VMD is enabled, this is
+		// performed within the script prepare method.
+		if err := sb.script.Reset(&req); err != nil {
+			return resp, errors.Wrap(err, "un-binding ssds")
 		}
```

Comment:
Wonder if it would be a better fit to move the VMD reset code into the Reset method. Then we could call Reset unconditionally here.

Reply:
The VMD "reset" isn't actually calling the script with the "reset" arguments, so it's not actually a reset and I think it would be confusing to add it in the script Reset method. The VMD initial call uses the setup script without an argument but just supplies the DRIVER_OVERRIDE=none envar.

Reply:
added Unbind script method along those lines